File: app/src/main/java/net/theluckycoder/advancedreboot/RebootTask.java
package net.theluckycoder.advancedreboot;

import android.os.AsyncTask;

import eu.chainfire.libsuperuser.Shell;

final class RebootTask extends AsyncTask<Void, Void, Void> {

    private final String[] mCommands;

    RebootTask(String[] commands) {
        mCommands = commands;
    }

    @Override
    protected Void doInBackground(Void... params) {
        Shell.SU.run(mCommands);
        return null;
    }
}

File: app/src/main/java/net/theluckycoder/advancedreboot/RebootTask.kt
package net.theluckycoder.advancedreboot

import eu.chainfire.libsuperuser.Shell
import java.util.concurrent.Executors

internal object RebootTask {

    private val executor = Executors.newSingleThreadExecutor()

    fun execute(commands: List<String>) {
        executor.execute {
            Shell.SU.run(commands)
        }
    }
}
